README.md

```diff
@@ -24,6 +25,48 @@ An operator can dynamically update a Node's role by promoting a Worker to Manage
+  - **Service Types**: There are different types of services. The project currently ships with two of them out of the box:
+    - **Replicated Services** can be scaled to a given number of replicas.
+    - **Global Services** will run one task on every node in the cluster.
+  - **Configurable Updates**: Every field of a service can be changed at any time.
+    After making an update, *SwarmKit* will reconcile the desired state by ensuring all tasks are using the desired settings.
+    By default, it will perform a lockstep update - that is, update all tasks at the same time. This can be configured through
+    different knobs:
+    - **Parallelism** defines how many updates can be performed at the same time
+    - **Delay** sets the minimum delay between updates. *SwarmKit* will start by shutting down the previous task, bring up a new one,
+      wait for it to transition to the *RUNNING* state *then* wait for the additional configured delay.
+      Finally, it will move onto other tasks.
+  - **Restart Policies**: The orchestration layer monitors tasks and reacts to failures based on the specified policy.
+    The operator can define restart conditions, delays and limits (maximum number of attempts in a given time window).
+    *SwarmKit* can decide to restart a task on a different machine. This means that faulty nodes will gradually be drained of their
+    tasks.
+- **Scheduling**
+  - **Resource Awareness**: *SwarmKit* is aware of resources available on nodes and will place tasks accordingly.
+  - **Constraints**: Operators can limit the set of nodes where a task can be scheduled by defining constraint expressions.
+    Constraints can match various node attributes such as IDs, names and labels (e.g. `node.labels.foo!=bar1`).
+  - **Strategies**: The project currently ships with a *spread strategy* which will attempt to schedule tasks on the least loaded
+    nodes, provided they meet the constraints and resource requirements.
+- **Cluster Management**
+  - **State Store**: Manager nodes maintain a strongly consistent, replicated (Raft based) and extremely fast (in-memory reads)
+    view of the cluster which allows them to make quick scheduling decisions while tolerating failures.
+  - **Topology Management**: Node roles (*Worker* / *Manager*) can be dynamically changed through API/CLI calls.
+  - **Node Management**: An operator can alter the desired availability of a node: Setting it to *Paused* will prevent any further
+    tasks from being scheduled to it while *Drained* will have the same effect while also re-scheduling its tasks somewhere else
+    (mostly for maintenance scenarios).
+- **Security**
+  - **Mutual TLS**: All nodes communicate with each other using mutual *TLS*. Swarm managers act as a *Root Certificate Authority*,
+    issueing certificates to new nodes.
+  - **Acceptance Policy**: Nodes joining a *Swarm* need to be accepted by an operator. Policies can be put in place to auto accept
```

Comment:
Policies can be put in place to auto accept, manually accept, or require a secret to join the cluster. Workers and Managers can have distinct join policies.